packages/slice-machine/pages/changes.tsx

```diff
@@ -80,6 +89,12 @@ const changes: React.FunctionComponent = () => {
 
   return (
     <Container sx={{ display: "flex", flex: 1 }}>
+      {unSyncedSlices.length ? (
+        <ScreenshotChangesModal
```

Comment:
The modal is not cross page. This is like saying that a Button is cross page because it's used in several places. The Modal _is_ passed an array of slices which _is_ the context it needs to know about

Reply:
Like i said, i think the modal should be global, have a single instance.

That way
- it can be initialised using the event to open it, passing Slices in the payload & all the parameters needed required.
- modals are grouped together and easier to find / interact with.
<img width="502" alt="image" src="https://user-images.githubusercontent.com/41187899/193248897-9a447e46-11f5-4908-aabd-4d6539191143.png">

We want this modal to be usable on many pages with different context so to avoid complexity in the code of each pages passing down functions to open the modal, initialising the modal with different state that changes, ... I would definitely makes it global and awaiting to be fed with data to open.

Reply:
PS: i'd use SliceIDs in the payload to always have Slices up to dates (ex: when a screenshot is taken on another tab to be updated).

Reply:
I agree with @hypervillain about the context though you’re right @Heyrows that you could have one instance of a modal in your app. You can’t implement it that simply, but could rely on tools like portals for instance.
I’d suggest here to consider using tools like Radix that will handle this kind of behaviour for you as well as keyboard controls, closing the modal when needed and so on.
WDYT?

Reply:
I'm going for an in-between where the SliceBuilder has one instance of modal and the Changes page too, and switches context based on user click. You can't really see that now because I did not code the SliceCards part. See this example (I used it to test if using functions in `useState` could give us free updates but I might do things more traditionally)

Reply:
The biggest problem with passing data via payload is that you don't get "free" updates that you get by passing them as props

Reply:
Seems like a good in-between for now and true about props.